game/resources/sfx.lua

```diff
@@ -0,0 +1,49 @@
+local Sfx = {}
+
+local Source = Class{}
+
+function Source:init(path, polyphony)
+    assert(polyphony > 0, "Not a valid polyphony value")
+    self.buffer_size = polyphony
+
+    self.buffer = {}
+    for i = 1, self.buffer_size do
+      self.buffer[i] = love.audio.newSource(path, "static")
+    end
+
+    self.next_index = 1
+end
+
+function Source:play()
+    self.buffer[self.next_index]:play()
+    self.next_index = (self.next_index % self.buffer_size) + 1
```

Comment:
I think this should be cached in the object

Reply:
I'm sorry, I didn't understand the comment haha can you explain it with other words?

Reply:
We should store the current index/last index instead of recalculating all the time. It makes things clearer.

Reply:
We are storing the current index, but okay, we can also store the last index. But this won't change this line you are commenting, after playing a sfx we will have to update them, and that is the only time we have to update them I think.

Reply:
Yeah I'm sorry I think I highlighted the wrong line. What I meant is that we do both the 'get next index' operation AND the 'get last index' operation. Ideally we want only to do one of them, if we can avoid it